From 78745bda5de0beed0411c4b1baf3783d84090024 Mon Sep 17 00:00:00 2001
From: Trung Tran <trung.tran@ettus.com>
Date: Wed, 15 Aug 2018 17:50:23 -0700
Subject: [PATCH] mpm: mg: move init_rf_cal before JESD de/framer bringup

This sequence is the one as described by the AD9371 user guide.
---
 mpm/python/usrp_mpm/dboard_manager/mg_init.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mpm/python/usrp_mpm/dboard_manager/mg_init.py b/mpm/python/usrp_mpm/dboard_manager/mg_init.py
index 213391655..99caf6d46 100644
--- a/mpm/python/usrp_mpm/dboard_manager/mg_init.py
+++ b/mpm/python/usrp_mpm/dboard_manager/mg_init.py
@@ -460,6 +460,10 @@ class MagnesiumInitManager(object):
         time.sleep(0.001) # 17us... ish.
         jesdcore.send_sysref_pulse()
         async_exec(self.mykonos, "finish_initialization")
+        # According to the AD9371 user guide, p.57, the RF cal must come before
+        # the framer/deframer init. We tried otherwise, and failed. So don't
+        # move this anywhere else.
+        self.init_rf_cal(args)
         self.log.trace("Starting JESD204b Link Initialization...")
         # Generally, enable the source before the sink. Start with the DAC side.
         self.log.trace("Starting FPGA framer...")
@@ -617,9 +621,5 @@ class MagnesiumInitManager(object):
                 "enabled inside Mykonos!")
             self.mykonos.enable_jesd_loopback(1)
         else:
-            # Now initialize calibrations:
-            # TODO: This also takes a long time. It might be faster to somehow
-            # just reset the calibrations, but one thing at a time.
-            self.init_rf_cal(args)
             self.mykonos.start_radio()
         return True
-- 
2.11.0

